cmd/kubeadm/app/cmd/upgrade/plan.go

```diff
@@ -92,85 +156,63 @@ func runPlan(flags *planFlags, userVersion string) error {
 
 	// Compute which upgrade possibilities there are
 	klog.V(1).Infoln("[upgrade/plan] computing upgrade possibilities")
-	availUpgrades, err := upgrade.GetAvailableUpgrades(versionGetter, flags.allowExperimentalUpgrades, flags.allowRCUpgrades, etcdClient, cfg.DNS.Type, client)
+	availUpgrades, err := upgrade.GetAvailableUpgrades(versionGetter, flags.allowExperimentalUpgrades, flags.allowRCUpgrades, etcdClient, cfg.DNS.Type, client, printer)
 	if err != nil {
 		return errors.Wrap(err, "[upgrade/versions] FATAL")
 	}
 
 	// Tell the user which upgrades are available
-	printAvailableUpgrades(availUpgrades, os.Stdout, isExternalEtcd)
-	return nil
+	return printAvailableUpgrades(availUpgrades, os.Stdout, isExternalEtcd, printer)
 }
 
 // printAvailableUpgrades prints a UX-friendly overview of what versions are available to upgrade to
-// TODO look into columnize or some other formatter when time permits instead of using the tabwriter
-func printAvailableUpgrades(upgrades []upgrade.Upgrade, w io.Writer, isExternalEtcd bool) {
+func printAvailableUpgrades(upgrades []upgrade.Upgrade, writer io.Writer, isExternalEtcd bool, printer output.Printer) error {
 
 	// Return quickly if no upgrades can be made
 	if len(upgrades) == 0 {
-		fmt.Fprintln(w, "Awesome, you're up-to-date! Enjoy!")
-		return
+		printer.Fprintf(writer, "Awesome, you're up-to-date! Enjoy!\n")
```

Comment:
We should probably add `printer.Fprintln`.

Reply:
ok, will do